### DISCUSSION TITLE
Is it possible to re-mount a virtual disk that has been unmounted but not detached?

### DISCUSSION THREAD

**Discussion by DarthRainbows** [LTRData/ImDisk#23] [Q&A] [answered]
I got myself into a bit of a quagmire with a virtual disk that contains important work-in-progress* that I would prefer not to lose if I don't have to. I accidentally ran the script that creates the virtual disk again, and instead of erroring out with a "there's a virtual disk mounted here already" message, imdisk just happily unmounted the old virtual disk, created a new one, and mounted it. The old one is still resident in memory (verified with `imdisk -l`, device 0 is the old drive). I detached the new drive, and I'd like to re-attach the old one, but I cannot figure out how to do so.

I've tried editing a virtual drive (`imdisk -e -u 0 -m z:`; just throws up the help text) and attaching a drive specifying the device (`imdisk -a -u 0 -m z:`; fails with `Error creating virtual disk: The parameter is incorrect.`). I don't see any other options that look likely to be what I need. Anyone know if this is possible to do and if so, how to do it?

* I know, not best practice to use a virtual disk store important work, but in this case, it's all temporary files that have heavy writes that I'd rather not have eating up write cycles on my SSD.

**Comment by Olof-Lagerkvist** [comment, marked as answer]
You could use my `dosdev` tool to assign a drive letter temporarily.

`dosdev -r Z: \Device\ImDisk0`

The tool is available on the website:
https://ltr-data.se/opencode.html

**Comment by TITANBGG** [comment]
it is possible

VMware
VMware Management Interface (vSphere or Workstation):

Go to the virtual machine settings.
In the Disks or Hard Disk section, you will see a list of virtual disks.
If a disk is detached, look for an option to reattach it in this list.
If necessary, manually select the disk and reattach it.
Command Line:

Use the vmware-vdiskmanager tool to check the disk.
Verify the location of the disk and reattach it to the correct virtual machine.
Hyper-V
Hyper-V Manager Console:
Go to the Settings of the virtual machine.
You will find virtual disks under SCSI Controller or IDE Controller.
To reattach a disconnected disk, click Browse and select the disk to attach it again.
VirtualBox
VirtualBox Management Interface:
Select your virtual machine and go to Settings.
In the Storage tab, you will see the virtual disks.
If the disk is disconnected, you can add a new disk or reattach the old one under the Controller (IDE or SATA).

**Reply by Olof-Lagerkvist** [reply]
Sorry, while your answer is probably correct and helpful for certain situations, it is not relevant to the scenario discussed in this thread.
